scripts/install/jellyfin.sh

```diff
@@ -0,0 +1,298 @@
+        <string>The Open Movie Database</string>
+        <string>TheMovieDb</string>
+      </DisabledMetadataFetchers>
+      <MetadataFetcherOrder />
+      <DisabledImageFetchers>
+        <string>The Open Movie Database</string>
+        <string>TheMovieDb</string>
+      </DisabledImageFetchers>
+      <ImageFetcherOrder />
+    </MetadataOptions>
+  </MetadataOptions>
+  <EnableAutomaticRestart>true</EnableAutomaticRestart>
+  <SkipDeserializationForBasicTypes>false</SkipDeserializationForBasicTypes>
+  <WanDdns>https://${ip_address}:${app_port_https}</WanDdns>
+  <UICulture>en-US</UICulture>
+  <SaveMetadataHidden>false</SaveMetadataHidden>
+  <ContentTypes />
+  <RemoteClientBitrateLimit>0</RemoteClientBitrateLimit>
+  <EnableFolderView>false</EnableFolderView>
+  <EnableGroupingIntoCollections>false</EnableGroupingIntoCollections>
+  <DisplaySpecialsWithinSeasons>true</DisplaySpecialsWithinSeasons>
+  <LocalNetworkSubnets />
+  <LocalNetworkAddresses>
+    <string>0.0.0.0</string>
+  </LocalNetworkAddresses>
+  <CodecsUsed />
+  <IgnoreVirtualInterfaces>false</IgnoreVirtualInterfaces>
+  <EnableExternalContentInSuggestions>true</EnableExternalContentInSuggestions>
+  <RequireHttps>false</RequireHttps>
+  <IsBehindProxy>false</IsBehindProxy>
+  <EnableNewOmdbSupport>false</EnableNewOmdbSupport>
+  <RemoteIPFilter />
+  <IsRemoteIPFilterBlacklist>false</IsRemoteIPFilterBlacklist>
+  <ImageExtractionTimeoutMs>0</ImageExtractionTimeoutMs>
+  <PathSubstitutions />
+  <EnableSimpleArtistDetection>true</EnableSimpleArtistDetection>
+  <UninstalledPlugins />
+</ServerConfiguration>
+CONFIG
+
+# Create the service file that will start and stop jellyfin.
+cat >"/etc/systemd/system/jellyfin.service" <<-SERVICE
+[Unit]
+Description=Jellyfin
+After=network.target
+[Service]
+User=${username}
+Group=${username}
+UMask=002
+Type=simple
+WorkingDirectory=/home/${username}/.jellyfin
+ExecStart=/home/${username}/.jellyfin/jellyfin -d /home/${username}/.config/Jellyfin
+TimeoutStopSec=20
+KillMode=process
+Restart=always
+RestartSec=2
+[Install]
+WantedBy=multi-user.target
+SERVICE
+
+# If DLNA port already in use disable DLNA
+if lsof -Pi :1900 -t >/dev/null; then
+  cat >"/home/${username}/.config/Jellyfin/config/dlna.xml" <<-CONFIG
+<?xml version="1.0"?>
+<DlnaOptions xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+  <EnablePlayTo>false</EnablePlayTo>
+  <EnableServer>false</EnableServer>
+  <EnableDebugLog>false</EnableDebugLog>
+  <BlastAliveMessages>true</BlastAliveMessages>
+  <SendOnlyMatchedHost>true</SendOnlyMatchedHost>
+  <ClientDiscoveryIntervalSeconds>60</ClientDiscoveryIntervalSeconds>
+  <BlastAliveMessageIntervalSeconds>1800</BlastAliveMessageIntervalSeconds>
+</DlnaOptions>
+CONFIG
+fi
+
+# Configure the nginx proxypass using positional parameters.
+if [[ -f /install/.nginx.lock ]]; then
+  bash "/usr/local/bin/swizzin/nginx/jellyfin.sh" "${app_port_http}" "${app_port_https}"
```

Comment:
nginx scripts should not be designed to be need arguments to build the correct configuration. This is important for the `box upgrade nginx` script (which also doubles as an nginx conf repair) as the final step of the repair process is to re-run all nginx scripts for installed apps, and it does so without arguments.

Thus, the nginx script should be crafted in a way that it detects the info it needs and pulls it automatically

Reply:
I am pretty sure the file this was based on already dealt with this:

https://github.com/userdocs/swizzin/blob/master/scripts/nginx/jellyfin.sh